app/components/notification_banner/view.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module NotificationBanner
+  class View < GovukComponent::Base
+    attr_reader :text, :title_id
+
+    def initialize(
+      title_text: nil,
+      text: nil,
+      type: nil,
+      classes: [],
+      role: nil,
+      title_id: nil,
+      disable_auto_focus: false
+    )
+      super(classes: classes)
+      @title_text = title_text
+      @text = text
+      @type = type
+      @role = role
+      @title_id = title_id || "govuk-notification-banner-title"
+      @disable_auto_focus = disable_auto_focus
+    end
+
+  private
+
+    attr_reader :title_text, :role, :type, :disable_auto_focus
+
+    def default_classes
+      %w[govuk-notification-banner]
+    end
+
+    def type_class
+      "govuk-notification-banner--#{type}" if success_banner?
+    end
+
+    def title
+      return title_text if title_text
+      return "Success" if success_banner?
```

Comment:
Might be a good idea to put these "success" "alert" type of strings into constants within the component if they're going to be asserted against?